test-util/src/main/java/io/zeebe/test/util/bpmn/random/ScheduledExecutionStep.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.test.util.bpmn.random;
+
+import io.zeebe.test.util.bpmn.random.steps.AbstractExecutionStep;
+import java.time.Duration;
+import java.util.Map;
+
+/**
+ * Class that schedules an execution step to run at a certain time.
+ *
+ * <p>"Time" here means a scheduled execution time.
+ *
+ * <p>To summarize, we distinguish the following points in time: _scheduledActivationTime_ this is
+ * the time when a BPMN element is activated.
+ *
+ * <p>An element might be activated, but its execution step(s) might be scheduled later. E.g. a
+ * parallel gateway will activate 2 or more elements, but only once of them will be executed next.
+ * The others are executed later. This is captured in
+ *
+ * <p>_scheduledExecutionStartTime_ this is the time when an execution step will be executed.
+ *
+ * <p>_deltaTime_ the time the execution step will take
+ *
+ * <p>_scheduledExecutionEndTime_ this is the time when the execution of a step is finished.
+ *
+ * <p>Distinguishing the activation vs. the execution time is important for calculating correct
+ * values for timeouts.
+ *
+ * <p>For a broader discussion, please consult https://github.com/camunda-cloud/zeebe/issues/6568
+ * which discusses the problem and the concept.
+ */
+public class ScheduledExecutionStep {
+
+  private final AbstractExecutionStep step;
+  private final ScheduledExecutionStep logicalPredecessor;
+
+  private final Duration activationTime;
+  private final Duration startTime;
+  private final Duration endTime;
+  private final Duration activationDuration;
+
+  protected ScheduledExecutionStep(
+      final ScheduledExecutionStep logicalPredecessor,
+      final ScheduledExecutionStep executionPredecessor,
+      final AbstractExecutionStep step) {
+    this.logicalPredecessor = logicalPredecessor;
+    this.step = step;
+
+    if (executionPredecessor != null) {
+      startTime = executionPredecessor.getEndTime();
+    } else {
+      startTime = Duration.ofMillis(0);
+    }
```

Comment:
Yes, duration for start and end time refers to the duration since the start of the test. I could also use ints here, but I like that Duration has a unit.